test/functional/api/v2/discovered_hosts_controller_test.rb

```diff
@@ -201,4 +201,14 @@ def test_auto_provision_all_no_rule_success
     assert_response :success
   end
 
+  def test_reboot_all_failure
+    disable_orchestration
+    host = discover_host_from_facts(@facts)
+    ::ForemanDiscovery::NodeAPI::PowerService.any_instance.expects(:reboot).returns(false)
+    post :reboot_all, params: { }
+    show_response = ActiveSupport::JSON.decode(@response.body)
+    assert_equal "Errors during reboot: #{host.name}: failed to reboot", show_response["error"]["message"]
+    assert_equal "macaabbccddeeff: failed to reboot ", show_response["error"]["host_details"][0]["error"]
+  end
```

Comment:
Can you rather assert whole JSON, so we can see how it looks like? Getting overall picture would be beneficial.